eventuals/catch.h

```diff
@@ -42,6 +59,85 @@ struct _Catch {
       e->Start();
     }
 
+    template <
+        typename Error_,
+        typename CatchHandler_,
+        typename... Tail>
+    void UnpackInvokeHelper(
```

Comment:
We need to handle the case that `Error` is actually a `std::exception_ptr` and introduce an explicit `try` that catches the type we want. Shall we chat about this synchronously?